gems/aws-sdk-core/lib/aws-sdk-core/rpc_v2/handler.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Aws
+  module RpcV2
+    class Handler < Seahorse::Client::Handler
+      # @param [Seahorse::Client::RequestContext] context
+      # @return [Seahorse::Client::Response]
+      def call(context)
+        build_request(context)
+        response = @handler.call(context)
+        response.on(200..299) { |resp| resp.data = parse_body(context) }
```

Comment:
Before we parse the body, we need to check for existence of `smithy-protocol` header in the response. If this does not have the same header, it must be considered malformed. Is this handled somewhere else in the core? I may have missed it.

Reply:
Yes. It's not handled yet, there's more to do.